packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -543,18 +562,11 @@ - (void)setPlaybackSpeed:(double)speed {
 }
 
 - (CVPixelBufferRef)copyPixelBuffer {
-  CVPixelBufferRef buffer = NULL;
-  CMTime outputItemTime = [_videoOutput itemTimeForHostTime:CACurrentMediaTime()];
-  if ([_videoOutput hasNewPixelBufferForItemTime:outputItemTime]) {
-    buffer = [_videoOutput copyPixelBufferForItemTime:outputItemTime itemTimeForDisplay:NULL];
-  } else {
-    // If the current time isn't available yet, use the time that was checked when informing the
-    // engine that a frame was available (if any).
-    CMTime lastAvailableTime = self.frameUpdater.lastKnownAvailableTime;
-    if (CMTIME_IS_VALID(lastAvailableTime)) {
-      buffer = [_videoOutput copyPixelBufferForItemTime:lastAvailableTime itemTimeForDisplay:NULL];
-    }
-  }
+  __block CVPixelBufferRef buffer = NULL;
+  dispatch_sync(self.pixelBufferSynchronizationQueue, ^{
+    buffer = self.frameUpdater.latestPixelBuffer;
+    self.frameUpdater.latestPixelBuffer = NULL;
```

Comment:
Since this pair is the only way the buffer should be consumed, a method on the frame updater that does both would be better than making the buffer publicly writeable.